contracts/trustless-rewards.clar

```diff
@@ -31,139 +34,105 @@
 (define-private (add-balance (id uint) (participant principal) (amount uint))
   (let ((lobby (unwrap! (map-get? lobbies id) (ok false))))
     (try! (stx-transfer? amount participant (as-contract tx-sender)))
-    (map-set lobbies id (merge lobby {balance: (+ (get balance lobby) amount)}))
-    (ok true)
-  )
-)
-
-
+    (map-set lobbies id (merge lobby {values: {balance: (+ (get balance (get values lobby)) amount), active: true, price: (get price (get values lobby))}}))
```

Comment:
Does this make it more clear what data is changed? 
```
(map-set lobbies id (merge lobby {values: (merge (get values lobby) {balance: (+ (get balance (get values lobby)) amount)})}))
```

Maybe the original solution with one big tuple make more sense.